### PR TITLE
[FIX] account: Check fiscal position on delivery

### PR DESCRIPTION
**Pull request by rousseldenis**
This is intended to show incoherence after this : https://github.com/odoo/odoo/pull/73831

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by rousseldenis** [comment]
@william-andre @epictete

File: addons/account/tests/test_fiscal_position.py [hunk 1]
```diff
@@ -154,3 +154,14 @@ def test_20_fp_one_tax_2m(self):
         mapped_taxes = self.fp2m.map_tax(self.src_tax)
 
         self.assertEqual(mapped_taxes, self.dst1_tax | self.dst2_tax)
+
+    def test_30_fp_country_delivery(self):
+        def assert_fp(partner, delivery, expected_pos, message):
+            self.assertEqual(
+                self.fp.get_fiscal_position(partner.id, delivery.id).id,
+                expected_pos.id,
+                message)
```
Comment (william-andre) [inline addons/account/tests/test_fiscal_position.py:163]: Do not copy paste

**Review by william-andre — COMMENTED** [review]
Can you also change the code; not only the test?

This should be good:

```diff
diff --git a/addons/account/models/partner.py b/addons/account/models/partner.py
index d95bf73002c..a339c94c56a 100644
--- a/addons/account/models/partner.py
+++ b/addons/account/models/partner.py
@@ -171,7 +171,7 @@ class AccountFiscalPosition(models.Model):
         delivery = PartnerObj.browse(delivery_id)
 
         # If partner and delivery have the same vat prefix, use invoicing
-        if not delivery.vat or (partner.vat and delivery.vat[:2] == partner.vat[:2]):
+        if delivery.vat and partner.vat and delivery.vat[:2] == partner.vat[:2]:
             delivery = partner
 
         # partner manually set fiscal position always win
```

File: addons/account/tests/test_fiscal_position.py (hunk 1 above)
Comment (rousseldenis) [reply addons/account/tests/test_fiscal_position.py:163]: Did it fast. Inspired by your own code :-)

**Comment by rousseldenis** [comment]
Ok, I will adapt

**Comment by rousseldenis** [comment]
@william-andre Done

File: addons/account/models/partner.py [hunk 2]
```diff
@@ -171,7 +171,7 @@ def get_fiscal_position(self, partner_id, delivery_id=None):
         delivery = PartnerObj.browse(delivery_id)
 
         # If partner and delivery have the same vat prefix, use invoicing
-        if not delivery.vat or (partner.vat and delivery.vat[:2] == partner.vat[:2]):
+        if not delivery or (delivery.vat and partner.vat and delivery.vat[:2] == partner.vat[:2]):
```
Comment (rousseldenis) [inline addons/account/models/partner.py:174]: Nearly. We should test if delivery is set first.

**Comment by rousseldenis** [comment]
@william-andre :green_apple:

File: addons/account/models/partner.py (hunk 2 above)
Comment (sbidoul) [reply addons/account/models/partner.py:174]: @rousseldenis what do you mean with this comment ?

Comment (rousseldenis) [reply addons/account/models/partner.py:174]: towards first @william-andre suggestion (see above)

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**